File: p49.hs
import Euler
import Data.List

kuwa = map (\x -> map toDigitInt (permutations x)) (combination [1..9] 4)

taka [] = 0
taka (x:xs)
    | isPrime x = 1 + taka xs
    | not (isPrime x) = taka xs

taka' [] = []
taka' (x:xs)
    | isPrime x = x : taka' xs
    | not (isPrime x) = taka' xs

ponya = map (sort . taka') $ filter (\x -> (taka x) >= 2) kuwa

ken xs = any (\x -> elem (div (sum x) 2) xs) (combination xs 2)

tarou = filter ken ponya

File: p49.py

def numToList(n):
    r = []
    while (n > 0):
        r.append(n % 10)
        n = n // 10
    r.sort()
    return r

def isPrime(n): # n is odd only.
    if n % 2 == 0:
        return False
    for i in range(3, n, 2):
        if n % i == 0:
            return False
    return True

for i in range(1001, 10000, 2):
    for j in range(i + 2, 10000, 2):
        k = (i + j) // 2
        if numToList(i) == numToList(j) and numToList(i) == numToList(k):
            if isPrime(i) and isPrime(j) and isPrime(k):
                print(i, k, j)

